packages/lib-sourcify/src/Validation/SolidityMetadataContract.ts

```diff
@@ -0,0 +1,383 @@
+import {
+  InvalidSources,
+  ISolidityCompiler,
+  IVyperCompiler,
+  Metadata,
+  MissingSources,
+  PathContent,
+  StringMap,
+  VyperJsonInput,
+  JsonInput,
+  Settings,
+  Libraries,
+} from '..';
+import { id as keccak256str } from 'ethers';
+import semver from 'semver';
+import { getIpfsGateway, performFetch } from './fetchUtils';
+
+const CONTENT_VARIATORS = [
+  (content: string) => content.replace(/\r?\n/g, '\r\n'),
+  (content: string) => content.replace(/\r\n/g, '\n'),
+];
+
+const ENDING_VARIATORS = [
+  (content: string) => content.trimEnd(),
+  (content: string) => content.trimEnd() + '\n',
+  (content: string) => content.trimEnd() + '\r\n',
+  (content: string) => content + '\n',
+  (content: string) => content + '\r\n',
+];
+
+// Dummy Compilation class
+class Compilation {
+  compiler: ISolidityCompiler | IVyperCompiler;
+  compilerVersion: string;
+  compilationTarget: string;
+  jsonInput: JsonInput | VyperJsonInput;
+  constructor(
+    compiler: ISolidityCompiler | IVyperCompiler,
+    compilerVersion: string,
+    compilationTarget: string,
+    jsonInput: JsonInput | VyperJsonInput,
+  ) {
+    this.compiler = compiler;
+    this.compilerVersion = compilerVersion;
+    this.compilationTarget = compilationTarget;
+    this.jsonInput = jsonInput;
+  }
+}
+
+export class SolidityMetadataContract {
+  metadata: Metadata;
+  name: string;
+  path: string;
+  providedSources: PathContent[];
+  providedSourcesByHash: Map<string, PathContent>;
+  foundSources: StringMap;
+  missingSources: MissingSources;
+  invalidSources: InvalidSources;
+  unusedSourceFiles: string[];
+  metadata2provided: StringMap; // maps the file path as in metadata.sources to the path of the provided by the user. E.g. metadata can have "contracts/1_Storage.sol" but the user provided "/Users/user/project/contracts/1_Storage.sol"
```

Comment:
I would rename this property, at least removing the 2: I like `metadataToProvidedPathMap`